### DISCUSSION TITLE
Named argument usage

### DISCUSSION THREAD

**Discussion by nanaya** [laravel/framework#50736] [Q&A]
The documentation [states](https://laravel.com/docs/11.x/releases#named-arguments):
> We may choose to rename function arguments when necessary in order to improve the Laravel codebase. Therefore, using named arguments when calling Laravel methods should be done cautiously and with the understanding that the parameter names may change in the future

But then [the examples on the very same page](https://laravel.com/docs/11.x/releases#middleware) use named arguments on laravel methods.

Can I assume all the examples with named arguments on the docs to be stable? Like [here](https://laravel.com/docs/11.x/eloquent#upserts), [here](https://laravel.com/docs/11.x/eloquent-mutators#defining-a-mutator), [here](https://laravel.com/docs/11.x/collections#method-groupby), and [here](https://laravel.com/docs/11.x/events#event-discovery) to list a few. Is there a more specific guidelines on which ones can use named arguments safely? Or should I ignore the examples and use them without named arguments?

**Comment by Tamas-hi** [comment]
That is a good question, and I would also like to know. Basically what I am asking is that changed name arguments will be present in the migration guide or not.

**Comment by shaedrich** [comment]
Good question indeed. This is bugging me ever since I encountered that statement in the docs as people are constantly using named arguments in their PRs, so their changes will potentially be subject to the explained changes in the future, which could cause problems.
